### ISSUE TITLE
RFE: ability to trigger services on fs fill threshold

### ISSUE THREAD

**Issue by james-lawrence** [systemd/systemd#28346]
### Component

systemd

### Is your feature request related to a problem? Please describe

would like a way to trigger services that clear disk space due to caching when disk reaches a particular threshold.

### Describe the solution you'd like

something similar to .path or .timer services but for disk utilization. but conditions + a timer would also be reasonable.

### Describe alternatives you've considered

manually writing a service that polls disk space and then triggers the event. but it'd be nice to not need to manually poll.

### The systemd version you checked that didn't have the feature you are asking for

253

**Comment by bluca** [comment]
You can use ConditionIOPressure= and a timer https://www.freedesktop.org/software/systemd/man/systemd.unit.html#ConditionMemoryPressure=

**Comment by vcaputo** [comment]
Pretty sure he means disk space used, not IO utilization...  and I'm not aware of any builtin hooks for monitoring space free of storage devices.

**Comment by james-lawrence** [comment]
Yeah I mean disk space. IO utilization isn't the important for this bit. I was actually surprised systemd didn't have anything for disk util yet.

**Comment by poettering** [comment]
to my knowledge there's no notification mechanism for quota events in the linux kernel. That's something people would have to add first really.

**Comment by james-lawrence** [comment]
@poettering detecting disk fill is a fairly cheap operation; even without kernel support it'd be feasible to poll to monitor it. not arguing that a notification mechanism from the kernel wouldn't be better. just saying pragmatically it wouldn't necessarily be a blocker.

**Comment by james-lawrence** [comment]
@poettering going to necro bump this a bit. to point out that a very simple implementation using polling that runs every 30s consumes less than 8s of time over 3 days. I think this demonstrates a notification mechanism isn't necessary for this functionality to be implemented by systemd as an opt in.

**Comment by bluca** [comment]
We are most definitely not going to poll disks as that's blocking I/O. This needs a notification mechanism added to the quota events in the kernel.

**Comment by james-lawrence** [comment]
From the peanut gallery world view that seems a excessive viewpoint on a feature request for something thats opt in, is in the ms range per disk, and can be done entirely async. Want to expand on the technical reasons behind your reasonings and how it'd be a problem?
